bin/activate.bat

```diff
@@ -1,78 +1,44 @@
-@echo off
-REM Check for CONDA_EVS_PATH environment variable
-REM It it doesn't exist, look inside the Anaconda install tree
-if "%CONDA_ENVS_PATH%" == "" (
-set CONDA_ENVS_PATH="%~dp0..\envs"
-)
+@echo on
+setlocal
 
-set CONDA_NEW_NAME=%~1
+set CONDA_NEW_ENV=%~1
 
-if "%~2" == "" goto skiptoomanyargs
-    echo ERROR: Too many arguments provided
-    goto usage
+if "%2" == "" goto skiptoomanyargs
+    (echo Error: did not expect more than one argument.) 1>&2
+    exit /b 1
 :skiptoomanyargs
 
-if "%CONDA_NEW_NAME%" == "" set CONDA_NEW_NAME=%~dp0..\
+if not "%1" == "" goto skipmissingarg
+    :: Set env to root if no arg provided
+    set CONDA_NEW_ENV=root
+:skipmissingarg
 
-REM Search through paths in CONDA_ENVS_PATH
-REM First match will be the one used
+REM Use conda itself to figure things out
 
-for %%F in ("%CONDA_ENVS_PATH:;=" "%") do (
-    if exist "%%~F\%CONDA_NEW_NAME%\conda-meta" (
-       set CONDA_NEW_PATH=%%~F\%CONDA_NEW_NAME%
-       goto found_env
-    )
-)
+SET "CONDA_EXE=%~dp0\conda.bat"
```

Comment:
They both exist.  I think the key difference is that conda.bat might actually just contain code pointing to the root conda.bat, which then knows about conda.exe.  Conda can only exist at the root level, or else the concept of the root level is lost, and deactivating to some reference state is more ambiguous.